python/lsst/pipe/tasks/processBrightStars.py

```diff
@@ -0,0 +1,383 @@
+
+        Returns
+        -------
+        result :  `lsst.pipe.base.Struct`
+            Result struct with component:
+
+            - ``brightStarStamps``: ``bSS.BrightStarStamps``
+        """
+        self.log.info("Extracting bright stars from exposure %s", dataId)
+        # Extract stamps around bright stars
+        extractedStamps = self.extractStamps(inputExposure, refObjLoader=refObjLoader)
+        # Warp (and shift, and potentially rotate) them
+        self.log.info("Applying warp to %i star stamps from exposure %s",
+                      len(extractedStamps.starIms), dataId)
+        warpedStars = self.warpStamps(extractedStamps.starIms, extractedStamps.pixCenters)
+        brightStarList = [bSS.BrightStarStamp(stamp_im=warp,
+                                              gaiaGMag=extractedStamps.GMags[j],
+                                              gaiaId=extractedStamps.gaiaIds[j])
+                          for j, warp in enumerate(warpedStars)]
+        # Compute annularFlux and normalize
+        self.log.info("Computing annular flux and normalizing %i bright stars from exposure %s",
+                      len(warpedStars), dataId)
+        # annularFlux statistic set-up, excluding mask planes
+        statsControl = afwMath.StatisticsControl()
+        statsControl.setNumSigmaClip(self.config.numSigmaClip)
+        statsControl.setNumIter(self.config.numIter)
+        innerRadius, outerRadius = self.config.annularFluxRadii
+        statsFlag = afwMath.stringToStatisticsProperty(self.config.annularFluxStatistic)
+        brightStarStamps = bSS.BrightStarStamps.initAndNormalize(brightStarList,
+                                                                 innerRadius=innerRadius,
+                                                                 outerRadius=outerRadius,
+                                                                 imCenter=self.modelCenter,
+                                                                 statsControl=statsControl,
+                                                                 statsFlag=statsFlag,
+                                                                 badMaskPlanes=self.config.badMaskPlanes)
+        return pipeBase.Struct(brightStarStamps=brightStarStamps)
+
+    def runDataRef(self, dataRef):
+        """ Read in required calexp, extract and process stamps around bright
+        stars and write them to disk.
+
+        Parameters
+        ----------
+        dataRef : `lsst.daf.persistence.butlerSubset.ButlerDataRef`
+            Data reference to the calexp to extract bright stars from.
+        """
+        calexp = dataRef.get("calexp")
+        output = self.run(calexp, dataId=dataRef.dataId)
+        # Save processed bright star stamps
+        dataRef.put(output.brightStarStamps, "brightStarStamps")
+        return pipeBase.Struct(brightStarStamps=output.brightStarStamps)
+
+    def runQuantum(self, butlerQC, inputRefs, outputRefs):
```

Comment:
We absolutely can not merge this as is. You cannot assume that 1) a gen2 butler is available 2) this is run on a machine that has this file path. If this would merge, it would seem to work in some cases, and then would fail in ci and integration tests.

At this point I dont think we should be prioritizing getting things done with the mentality of defaulting to gen2 and then addressing gen3 later as a secondary concern. Gen2 is deprecated, so unless there is some technical reason something can not be done in gen3 yet, I feel this should be the bar for considering something "working". The only technical limitation I know of getting Gaia support is simply doing the work.

There are two paths forward I see. 1) work with the current bootstrapping to get Gaia calibrations migrated when the repo is converted to a gen3 repo. 2) Do the work to get Gaia ingested naively.

What you have below is fine for proving that you code could work, but it is not there yet for merging to master.